src/script/common/c_content.cpp

```diff
@@ -1959,8 +1959,18 @@ void push_objectRef(lua_State *L, const u16 id)
 
 void read_hud_element(lua_State *L, HudElement *elem)
 {
-	elem->type = (HudElementType)getenumfield(L, 2, "hud_elem_type",
-									es_HudElementType, HUD_ELEM_TEXT);
+	// Handle deprecated hud_elem_type
+	std::string type_string;
+	if (getstringfield(L, 2, "hud_elem_type", type_string)) {
+		log_deprecated(L, "Deprecated \"hud_elem_type\" field, use \"type\" instead.");
```

Comment:
This function might be called multiple times per server step. Please eliminate duplicates using `script_log_unique`.